contracts/HubPool.sol

```diff
@@ -47,10 +43,22 @@ contract HubPool is Testable, Lockable, MultiCaller, Ownable {
 
     RefundRequest public refundRequest;
 
-    // Whitelist of origin token to destination token routings to be used by off-chain agents.
+    // Whitelist of origin token to destination token routings to be used by off-chain agents. The notion of a route
+    // does not need to include L1; it can store L2->L2 routes i.e USDC on Arbitrum -> USDC on Optimism as a "route".
     mapping(address => mapping(uint256 => address)) public whitelistedRoutes;
 
-    mapping(address => LPToken) public lpTokens; // Mapping of L1TokenAddress to the associated LPToken.
```

Comment:
I redefined a fair bit of the internal structures. for each L1 token we need more information than just the `lpToken` and `isEnabled`. The struct below contains _all_ the info we need for a given "PooledToken".

Reply:
Note that this structural change rippled through this file and the unit tests.